addons/mail/controllers/thread.py

```diff
@@ -168,9 +168,14 @@ def _filter_message_post_partners(self, thread, partners):
     @add_guest_to_context
     def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
         guest = request.env["mail.guest"]._get_guest_from_context()
-        guest.env["ir.attachment"].browse(post_data.get("attachment_ids", []))._check_attachments_access(
-            kwargs.get("attachment_tokens")
-        )
+        if (
+            not guest.env["ir.attachment"]
```

Comment:
I don't think we need to use the guest env anymore, this was only useful when checking message authorship inside _check_attachments_access